packages/liveblocks-react/src/liveblocks.tsx

```diff
@@ -350,6 +350,17 @@ function makeLiveblocksExtrasForClient(client: OpaqueClient) {
 
   const userThreadsPoller = makePoller(async () => {
     try {
+      //
+      // XXX Think about this! To remain symmetric with the solid inbox
+      // XXX notifications implementation, we should first call something like:
+      //       await store.waitUntilUserThreadsLoaded(query);
+      //                                              ^^^^^
+      // XXX                         However... which query should we use here?
+      //
+      // XXX Maybe this warrents an API like? 🤔
+      //       await store.waitUntilAnyUserThreadsLoaded();
```

Comment:
Hmm. Here's my understanding of how this currently works:

A poller starts as soon as a `useUserThreads` hook mounts and the poller runs, say every 2 hours. Let's say the first time it runs, user threads for none of the query key has been successfully loaded yet. In that case, the poller does nothing because the poller doesn't know when to fetch user threads delta since, i.e. it doesn't have the `lastRequestedAt` date value. So, we hope the next time the poller run (in 2 hours), user threads for at least a query key has been successfully loaded and the poller can then start fetching updates with the some value of `lastRequestedAt`.

With the change you're proposing (to also keep the logic consistent with inbox notifications), each time the poller runs (every 2 hours), we'll wait until until user threads for at least one query is loaded (this will be instant if the condition is true), and only then do we request delta updates. With this behavior, what if no user threads have been loaded both the first and the second time the poller runs and say after a few seconds the user threads are loaded, then both requests that were waiting for user threads to be available for at least a query is complete will kick off.

Do we instead want to only start the poller only after user threads are loaded for at least a query key? Happy to discuss this in the call.

Reply:
Updating for posterity. This was discussed in yesterday's call, and we decided that we'll keep the current no-op behavior if the `lastRequestedAt` date is still unknown, but next week we'll want to give it a shot to clean this behavior up and not even make the poller start and make it fully dependent on the promise that needs to be fulfilled first (aka the moment the `lastRequestedAt` becomes set). Only _then_ we'll want to enable the poller.

Also we observed that two poll executions from the same poller can currently overlap, which is undesirable because of the possible race conditions. This isn't likely to happen in practice because the fetch requests are typically faster than the `POLLING_INTERVAL`, but there is a race condition there.

Ideally we'll fix this by making the following changes:
1. Set a max timeout for a single poll cycle (proposal POLLING_INTERVAL + some buffer).
2. Ensure we'll only schedule the next poll `POLLING_INTERVAL` after the polling function has been settled.

Current (possible) timeline:
```
|<- POLL_INTERVAL ->|
|-------------------|------------o------|-------------------|-------------------|-------------------|...
^ Mount             ^ Poll 1     ^      ^ Poll 2            ^ Poll 3    ^       ^ Poll 4   ^          ^
                      (noop)     |                                      |                  |          |
                            First page loaded                       Poll 2 ends      Poll 3 ends   Poll 4 ends
```

Proposed new behavior (researching/implementing next week):
```
                                 |<- POLL_INTERVAL ->|          |<- POLL_INTERVAL ->|
|--------------------------------o-------------------|----------|-------------------|----------|...
^ Mount                          ^                   |< Poll 1 >|                   |< Poll 2 >|
                                 |
                            First page loaded
```